### ISSUE TITLE
Fundamental overrides in `Node`, `Tank`, `DecayTank`, etc.

### ISSUE THREAD

**Issue by liuly12** [ImperialCollegeLondon/wsi#73]
This issue is set to discuss any needs for enabling overrides in fundamental nodes including `Node`, `Tank`, `DecayTank`, etc.

**Comment by barneydobson** [comment]
Do we need an issue for `Node` overrides? the only parameter is `name` which I think we don't allow to override

**Comment by liuly12** [comment]
How to override `decay`, `datum`, `initial_storage`? Should these overrides be put into the `Surface` or the `Tank`, `DecayTank`?

**Comment by barneydobson** [comment]
OK it makes sense - I guess `Tank` overrides are the sensible solution to this

**Comment by liuly12** [comment]
Do we want to handle `initial_storage` -> `storage`? If yes, how? I guess the `storage` cannot be a property because it is going to be updated at each timestep?
```
        # Set parameters
        self.capacity = capacity
        self.area = area
        self.datum = datum
        self.initial_storage = initial_storage

        WSIObj.__init__(self)  # Not sure why I do this rather than super()

        # TODO I don't think the outer if statement is needed
        if "initial_storage" in dir(self):
            if isinstance(self.initial_storage, dict):
                # Assume dict is VQIP describing storage
                self.storage = self.copy_vqip(self.initial_storage)
                self.storage_ = self.copy_vqip(
                    self.initial_storage
                )  # Lagged storage for mass balance
            else:
                # Assume number describes initial stroage
                self.storage = self.v_change_vqip(
                    self.empty_vqip(), self.initial_storage
                )
                self.storage_ = self.v_change_vqip(
                    self.empty_vqip(), self.initial_storage
                )  # Lagged storage for mass balance
        else:
            self.storage = self.empty_vqip()
            self.storage_ = self.empty_vqip()  # Lagged storage for mass balance
```

**Comment by barneydobson** [comment]
Well I think the main reason `initial_storage` was its own attribute was to enable saving of the model that reverts the `config` to its initial state.

I think it makes sense that we do not allow `storage` to be set in `overrides` - though could add a little catch that pings a message like "`storage` not override-able, please use `initial_storage`" or something.

I would suggest that we put this in a function:
```
            if isinstance(self.initial_storage, dict):
                # Assume dict is VQIP describing storage
                self.storage = self.copy_vqip(self.initial_storage)
                self.storage_ = self.copy_vqip(
                    self.initial_storage
                )  # Lagged storage for mass balance
            else:
                # Assume number describes initial stroage
                self.storage = self.v_change_vqip(
                    self.empty_vqip(), self.initial_storage
                )
                self.storage_ = self.v_change_vqip(
                    self.empty_vqip(), self.initial_storage
                )  # Lagged storage for mass balance
```

that is called by `apply_overrides` and the `__init__`? 

There may be a more elegant solution that makes use of `property` - but I think for now let's focus on making `overrides` work properly, if there are any further clever ideas you can always make them a new issue

**Comment by barneydobson** [comment]
fixed by #91
